pkg/cluster/cluster.go

```diff
@@ -0,0 +1,67 @@
+package cluster
+
+import (
+	"context"
+	"fmt"
+	"os"
+
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	distributionConfigMapName = "llama-stack-k8s-operator-distribution-images"
+)
+
+type ClusterInfo struct {
+	OperatorNamespace  string
+	DistributionImages map[string]string
+}
+
+var clusterInfo *ClusterInfo
+
+func Init(ctx context.Context, client client.Client) error {
```

Comment:
Would you add a brief description of the purpose of this function? It looks like it initializes a singleton `ClusterInfo`. Why a singleton?